File: internal/api/controllers/music/get-song.go
package music_controller

import (
	"jota/server/internal/repositories"

	"github.com/gofiber/fiber/v3"
)

func GetSong(c fiber.Ctx) error {
	id := c.Params("id")

	song, err := repositories.Use.Music.GetSong(id)
	if err != nil {
		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
			"error": err.Error(),
		})
	}

	return c.Status(fiber.StatusOK).JSON(song)
}

File: client/src/lib/music/app/get-song.ts
import type { Song } from "@/lib/music/model";
import { get } from "@/lib/shared/api";

export async function getSong(id: string): Promise<Song> {
	try {
		const response = await get<Song>(`/music/song/${id}`);
		return response;
	} catch (error) {
		console.error(error);
		throw error;
	}
}
